### PR TITLE
Issue warning for variables called `super` or `this`

### LINKED ISSUE #6617: Variables can be called `super` or `this`
```
contract Test {
	function f2() public pure returns (uint a, uint b ) {
		uint super = 3;
		uint this = 4;
		uint blockhash = 5; // unsure about whether this should be a warn/err
		uint gasleft = 7; // unsure about whether this should be a warn/err
		return (super, this);
	}
}
```

### PR DESCRIPTION
**Pull request by Marenz**
fixes #6617

### CODE REVIEW

File: libsolidity/interface/CompilerStack.cpp [hunk 1]
```diff
@@ -278,9 +278,6 @@ bool CompilerStack::analyze()
 			for (ASTPointer<ASTNode> const& node: source->ast->nodes())
 				if (ContractDefinition* contract = dynamic_cast<ContractDefinition*>(node.get()))
 				{
-					m_globalContext->setCurrentContract(*contract);
```
Comment (chriseth) [inline libsolidity/interface/CompilerStack.cpp:281]: Ah, that looks good! :)

File: libsolidity/interface/CompilerStack.cpp [hunk 2]
```diff
@@ -262,7 +262,7 @@ bool CompilerStack::analyze()
 		m_globalContext = make_shared<GlobalContext>();
 		NameAndTypeResolver resolver(m_globalContext->declarations(), m_scopes, m_errorReporter);
 		for (Source const* source: m_sourceOrder)
-			if (!resolver.registerDeclarations(*source->ast))
+			if (!resolver.registerDeclarations(*source->ast, *m_globalContext.get()))
```
Comment (chriseth) [inline libsolidity/interface/CompilerStack.cpp]:
```suggestion
			if (!resolver.registerDeclarations(*source->ast, *m_globalContext))
```

**Review by chriseth — DISMISSED** [review]

**Comment by chriseth** [comment]
Please add tests.

**Review by chriseth — DISMISSED** [review]

**Comment by chriseth** [comment]
Ah, please also add a changelog entry (for 0.5.9).

**Comment by Marenz** [comment]
added changelog

File: Changelog.md [hunk 3]
```diff
@@ -33,6 +33,7 @@ Bugfixes:
  * SMTChecker: SSA control-flow did not take into account state variables that were modified inside inlined functions that were called inside branches.
  * Type System: Use correct type name for contracts in event parameters when used in libraries. This affected code generation.
  * Type System: Allow direct call to base class functions that have overloads.
+ * Type System: Warn about shadowing builtin variables if user variables are named "this" or "super"
```
Comment (leonardoalt) [inline Changelog.md]:
```suggestion
 * Type System: Warn about shadowing builtin variables if user variables are named "this" or "super".
```

Comment (chriseth) [reply Changelog.md]: also maybe `this` and `super`.

**Comment by Marenz** [comment]
fixed changelog

**Review by chriseth — APPROVED** [review]

**Pull request merged**